### PR TITLE
Enable Codecov integration

### PR DESCRIPTION
**Pull request by fingolfin**
Also update some actions to their latest version.

Note that this may require additional admin action, e.g.
adding / enabling the Codecov integration from the GitHub
marketplace for this repository.

### CODE REVIEW

**Comment by fingolfin** [comment]
Let's re-run jobs based on lastest `master`

**Comment by fingolfin** [comment]
The log report an error, which is also visible at <https://app.codecov.io/github/JuliaRandom/RandomExtensions.jl>:

> This repo has been deactivated
> Contact an administrator of your git organization to grant write-permissions in your git-provider for this repository.

I suppose someone with the right access level has to go to <https://github.com/JuliaRandom/RandomExtensions/settings/installations> and verify the Codecov is there, and enabled for this repo.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -21,14 +21,14 @@ jobs:
           - x64
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (fingolfin) [inline .github/workflows/CI.yml:24]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/Documentation.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (fingolfin) [inline .github/workflows/Documentation.yml:14]:
```suggestion
      - uses: actions/checkout@v4
```

**Comment by fingolfin** [comment]
OK, https://app.codecov.io/gh/JuliaRandom/RandomExtensions.jl shows data again (go to the "Pulls" tab). Once this is merged (better: squashed) it should also show data again on the front, for `master`.

I'll not merge this immediately as @rfourquet wanted to check something -- maybe that was just about checking why it didn't work (which should be resolved now), but maybe there was something else?

**Comment by rfourquet** [comment]
>  but maybe there was something else?

Yeah I wanted to look more into this coverage topic, to understand the options. I still didn't find a lot of online resources about this, is codecov essentially about reporting coverage information on PRs? I can see the appeal, but for example in this very PR codecov comments are unrelated, so it seems noisy... I'm open to try it out though.

Are there other options people use for tracking coverage, beside the built-in `--code-coverage` CLI option? maybe Coveralls? I also saw Coverage.jl.

**Comment by fingolfin** [comment]
Codecov is the only one that reliably works in my experience. Coveralls exists but I never got it to work reliably nor have I seen it work reliably on other projects.

>  but for example in this very PR codecov comments are unrelated, so it seems noisy.

Well, yeah, they point out things that are not covered -- they do so because there is no baseline data for `master`... Once this is merged, I don't think this noise won't show up for future PRs. (That said, the issue is relevant: it points to code that is not triggered by any tests in this package(

I am not quite sure what you are concerned about, though -- in the worst case this can be removed again later? But to see useful data, this needs to be merged first (so there is a baseline), and then you can experience the effect on future PRs.

**Comment by rfourquet** [comment]
Ok thanks for your comments, let's try it out

**Pull request merged**